main.py

```diff
@@ -15,41 +15,77 @@
 return: string representing output code or error message
 """
 def main(input_code, input_lang, output_lang):
+    try:
+        # check arguments TODO(taiga#172): remove hard coded references
+        check_valid_args(input_code, input_lang, output_lang, ["js", "py", "java"], ["js", "py", "bash", "java"])
 
-    if input_lang == None and output_lang == None:
-        abort(400, "Error: must specify input and output languages")
+        # code to gast
+        gast = main_code_to_gast(input_code, input_lang)
+        check_valid_gast(gast)
 
-    if input_lang == "js":
-        gast = js_main.js_to_gast(input_code)
-    elif input_lang == "py":
-        gast = py_main.py_to_gast(input_code)
-    elif input_lang == "java":
-        gast = java_main.java_to_gast(input_code)
-    else:
-        #TODO: figure out hwo to do error messages
-        return "Error must specify input language. For example, js for javascript and py for python"
+        #gast to code
+        output_code = main_gast_to_code(gast, output_lang)
+
+        # analytics
+        main_analytics(input_code, output_code, input_lang, output_lang)
```

Comment:
Nit:
Describe the action in the name i.e. store_analytics to make it easier to understand what the function is doing without having to read the function body.

Reply:
good point! Will address